embedded_fab/main.dart

```diff
@@ -0,0 +1,416 @@
+      ),
+      BottomNavItem(
+        icon: Icons.archive,
+        isSelected: _selectedIndex == 3,
+        title: 'Archived',
+        onTap: () {
+          _onItemTapped(3);
+        },
+      )
+    ];
+  }
+
+  Widget _experimentalBottomNav() {
+    PlusPainter plusPainter;
+    switch (_selectedShape) {
+      case ShapeType.Bump:
+        {
+          plusPainter = PlusPainter(color: _strokeColor, yOffset: 36);
+          break;
+        }
+      case ShapeType.Cut:
+        {
+          plusPainter = PlusPainter(color: _strokeColor, yOffset: 24);
+          break;
+        }
+      case ShapeType.Flat:
+        {
+          plusPainter = PlusPainter(color: _strokeColor, iconSize: 24, yOffset: 62);
+          break;
+        }
+    }
+
+    return GestureDetector(
+      onTap: () {
+        _onFABTap();
+      },
+      child: CustomPaint(
+        foregroundPainter: plusPainter,
+        painter: ContainerPainter(
+          fillColor: _selectedColor,
+          strokeColor: _strokeColor,
+          shape: _selectedShape,
+        ),
+        child: SafeArea(
+          child: ConstrainedBox(
+            constraints: BoxConstraints(maxHeight: 92),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              children: [
+                SizedBox(height: 46),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: _itemOptions(),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _bottomAppBar() {
+    return BottomAppBar(
+      shape: GoogleBabShape(),
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 8.0),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            IconButton(icon: Icon(Icons.menu), onPressed: () {}),
+            IconButton(icon: Icon(Icons.search), onPressed: () {})
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _bottomNavSideFAB() {
+    return BottomNavigationBar(
+      items: [
+        BottomNavigationBarItem(
+          icon: Icon(Icons.inbox),
+          title: Text('Inbox'),
+        ),
+        BottomNavigationBarItem(
+          icon: Icon(Icons.star),
+          title: Text('Starred'),
+        ),
+        BottomNavigationBarItem(
+          icon: Icon(Icons.send),
+          title: Text('Sent'),
+        ),
+        BottomNavigationBarItem(
+          icon: Icon(Icons.archive),
+          title: Text('Archived'),
+        ),
+      ],
+      backgroundColor: Colors.white,
+      selectedItemColor: blue500,
+      unselectedItemColor: grey600,
+      currentIndex: _selectedIndex,
+      elevation: 5,
+      showUnselectedLabels: true,
+      type: BottomNavigationBarType.fixed,
+      onTap: (value) {
+        setState(() {
+          _selectedIndex = value;
+        });
+      },
+    );
+  }
+
+  void _onFABTap() {
```

Comment:
```suggestion
  void _onFabTap() {
```

`FAB` should be `Fab` or `fab` everywhere 

https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words

Reply:
Done